src/gateway/WebSocketClient.ts

```diff
@@ -0,0 +1,127 @@
+import WebSocket from 'ws';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { EVENTS, Payload } from './constants';
+import { GATEWAY_OPCODES, GATEWAY_CLOSE_EVENT_CODES } from '../util/Constants';
+
+export default class WebSocketClient {
+  private _intents: number;
+  private _token: string;
+  private _wsURL: string;
+  private heartBeatInterval!: NodeJS.Timeout;
+  private lastHeartbeatAck = true;
+  private seq!: number;
+  private session_id!: string;
+  private ws!: WebSocket;
+
+  constructor(token: string, wsURL: string, intents: number) {
+    this._token = token;
+    this._wsURL = wsURL;
+    this._intents = intents;
+
+    this.identify = this.identify.bind(this);
```

Comment:
Original review https://github.com/libraryofcode/dis.ts/pull/15#discussion_r548976462

Reply:
I think it's fine as it is

Reply:
yeah we'll keep it like that